test-clients/src/main/java/com/hedera/services/bdd/suites/token/TokenTransactSpecs.java

```diff
@@ -430,6 +446,189 @@ public HapiApiSpec balancesChangeOnTokenTransfer() {
+				);
+	}
+
+	public HapiApiSpec uniqueTokenTxnWithSenderNotSigned() {
+		return defaultHapiSpec("UniqueTokenTxnWithOwnerNotSigned")
+				.given(
+						newKeyNamed("supplyKey"),
+						newKeyNamed("signingKeyTreasury"),
+						newKeyNamed("signingKeyFirstUser"),
+						cryptoCreate(TOKEN_TREASURY).key("signingKeyTreasury"),
+						cryptoCreate(FIRST_USER).key("signingKeyFirstUser"),
+						tokenCreate(A_TOKEN)
+								.tokenType(TokenType.NON_FUNGIBLE_UNIQUE)
+								.supplyType(TokenSupplyType.INFINITE)
+								.initialSupply(0)
+								.supplyKey("supplyKey")
+								.treasury(TOKEN_TREASURY),
+						tokenAssociate(FIRST_USER, A_TOKEN)
+				)
+				.when(
+						mintToken(A_TOKEN, List.of(ByteString.copyFromUtf8("memo")))
+				)
+				.then(
+						cryptoTransfer(
+								movingUnique(1, A_TOKEN).between(TOKEN_TREASURY, FIRST_USER)
+						)
+								.signedBy(DEFAULT_PAYER)
+								.hasKnownStatus(INVALID_SIGNATURE)
+				);
+	}
+
+	public HapiApiSpec uniqueTokenTxnWithReceiverNotSigned() {
+		return defaultHapiSpec("UniqueTokenTxnWithOwnerNotSigned")
+				.given(
+						newKeyNamed("supplyKey"),
+						newKeyNamed("signingKeyTreasury"),
+						newKeyNamed("signingKeyFirstUser"),
+						cryptoCreate(TOKEN_TREASURY).key("signingKeyTreasury"),
+						cryptoCreate(FIRST_USER).key("signingKeyFirstUser").receiverSigRequired(true),
+						tokenCreate(A_TOKEN)
+								.tokenType(TokenType.NON_FUNGIBLE_UNIQUE)
+								.supplyType(TokenSupplyType.INFINITE)
+								.initialSupply(0)
+								.supplyKey("supplyKey")
+								.treasury(TOKEN_TREASURY),
+						tokenAssociate(FIRST_USER, A_TOKEN)
+				)
+				.when(
+						mintToken(A_TOKEN, List.of(ByteString.copyFromUtf8("memo")))
+				)
+				.then(
+						cryptoTransfer(
+								movingUnique(1, A_TOKEN).between(TOKEN_TREASURY, FIRST_USER)
+						)
+								.signedBy("signingKeyTreasury", DEFAULT_PAYER)
+								.hasKnownStatus(INVALID_SIGNATURE)
+				);
+	}
+
+	public HapiApiSpec uniqueTokenTxnsAreAtomic() {
+		return defaultHapiSpec("UniqueTokenTxnsAreAtomic")
+				.given(
+						newKeyNamed("supplyKey"),
+						newKeyNamed("signingKeyTreasury"),
+						newKeyNamed("signingKeyFirstUser"),
+						cryptoCreate(FIRST_USER).key("signingKeyFirstUser"),
+						cryptoCreate(SECOND_USER),
+						cryptoCreate(TOKEN_TREASURY).key("signingKeyTreasury"),
+						tokenCreate(A_TOKEN)
+								.tokenType(TokenType.NON_FUNGIBLE_UNIQUE)
+								.supplyType(TokenSupplyType.INFINITE)
+								.initialSupply(0)
+								.supplyKey("supplyKey")
+								.treasury(TOKEN_TREASURY),
+						tokenCreate(B_TOKEN)
+								.initialSupply(100)
+								.treasury(TOKEN_TREASURY),
+						mintToken(A_TOKEN, List.of(ByteString.copyFromUtf8("memo"))),
+						tokenAssociate(FIRST_USER, A_TOKEN),
+						tokenAssociate(FIRST_USER, B_TOKEN),
+						tokenAssociate(SECOND_USER, A_TOKEN)
+				)
+				.when(
+						mintToken(A_TOKEN, List.of(ByteString.copyFromUtf8("memo")))
+				)
+				.then(
+						cryptoTransfer(
```

Comment:
In order for this test to be best executed we need  to:
1) do the `movingUnique` before `moving` fungibles, because it fails on the transfer of fungibles thus it won't apply the non fungible changes because it reverts prematurely 
2) Assert that the balances of TOKEN A and TOKEN B have not changed

Reply:
It's done